src/transformStateWithClones.js

```diff
@@ -7,7 +7,36 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  let transState = { ...state };
+  const cloneState = [];
+
+  for (const i in actions) {
+    const { type, extraData, keysToRemove } = actions[i];
+
+    if (type === 'addProperties') {
+      for (const key in extraData) {
+        transState = Object.assign(transState, { [key]: extraData[key] });
+      }
+      cloneState.push(Object.assign({}, transState));
+    }
+
+    if (type === 'removeProperties') {
+      for (const key in keysToRemove) {
+        // transState = Object.assign(state, { [key]: keysToRemove[key] });
```

Comment:
This line is commented out and seems to be incorrect. If you intended to use it, ensure it correctly implements the intended logic. Otherwise, it should be removed.